pkg/config/configuration.go

```diff
@@ -44,9 +45,9 @@ const (
 )
 
 type Configuration struct {
-	Name string            `json:"-" yaml:"-"`
-	Kind string            `json:"-" yaml:"-"`
-	Spec ConfigurationSpec `json:"spec" yaml:"spec"`
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
```

Comment:
Add a comment linking to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata ?